packages/legacy/core/App/screens/PasteUrl.tsx

```diff
@@ -0,0 +1,139 @@
+import { useAgent } from '@credo-ts/react-hooks'
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View, TextInput, ScrollView, TouchableOpacity, Modal } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+
+import Button, { ButtonType } from '../components/buttons/Button'
+import InfoBox, { InfoBoxType } from '../components/misc/InfoBox'
+import { TOKENS, useContainer } from '../container-api'
+import { useConfiguration } from '../contexts/configuration'
+import { useTheme } from '../contexts/theme'
+import { ConnectStackParams } from '../types/navigators'
+import { connectFromScanOrDeepLink } from '../utils/helpers'
+import { testIdWithKey } from '../utils/testable'
+
+export type PasteProps = StackScreenProps<ConnectStackParams>
+
+const PasteUrl: React.FC<PasteProps> = ({ navigation }) => {
+  const { ColorPallet, TextTheme } = useTheme()
+  const [pastedContent, setPastedContent] = useState<string>('')
+  const [errorMessage, setErrorMessage] = useState<{ title: string; message: string } | undefined>()
+  const { t } = useTranslation()
+  const { agent } = useAgent()
+  const container = useContainer()
+  const logger = container.resolve(TOKENS.UTIL_LOGGER)
+  const { enableImplicitInvitations, enableReuseConnections } = useConfiguration()
+  const styles = StyleSheet.create({
+    container: {
+      flex: 1,
+      backgroundColor: ColorPallet.brand.primaryBackground,
+    },
+    content: {
+      margin: 20,
+    },
+    description: {
+      ...TextTheme.normal,
+      marginBottom: 20,
+    },
+    textBox: {
+      ...TextTheme.normal,
+      textAlignVertical: 'top',
+      borderColor: ColorPallet.grayscale.darkGrey,
+      borderWidth: 1,
+      borderRadius: 5,
+      backgroundColor: ColorPallet.grayscale.lightGrey,
+    },
+    buttonContainer: {
+      margin: 20,
+    },
+  })
+
+  const processPastedContent = async () => {
+    try {
+      await connectFromScanOrDeepLink(
+        pastedContent,
+        agent,
+        logger,
+        navigation?.getParent(),
+        false, // isDeepLink
+        enableImplicitInvitations,
+        enableReuseConnections
+      )
+    } catch (err: unknown) {
+      setErrorMessage({ title: t('PasteUrl.ErrorInvalidUrl'), message: t('PasteUrl.ErrorInvalidUrlDescription') })
+    }
+  }
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <ScrollView>
+        <Modal visible={!!errorMessage} testID={testIdWithKey('ErrorModal')} animationType="fade" transparent>
```

Comment:
A `DismissiblePopupModal` might be good here, it's scrollable incase the error is long

Reply:
Is there any way to make it look like an error popup instead of an info one? I'm not seeing anything in the component to support it

Reply:
Not currently, and if we're not including that `err.message` from above then it won't be variable length (won't need scroll), and so the InfoBox works as is. Approving